### PR TITLE
Support Application Log collecting

### PR DESCRIPTION
**Pull request by Oseenix**

### CODE REVIEW

**Comment by observeralone** [comment]
I see that many unit tests are not written, will you add them later?

**Comment by Oseenix** [comment]
> I see that many unit tests are not written, will you add them later?

Let's book an online code review today

File: plugin-api/src/main/java/io/opentelemetry/sdk/resources/EaseAgentResource.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAMESPACE;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class EaseAgentResource extends Resource implements ConfigChangeListener {
+    static volatile EaseAgentResource agentResource = null;
+
+    public EaseAgentResource() {
```
Comment (zhao-kun) [inline plugin-api/src/main/java/io/opentelemetry/sdk/resources/EaseAgentResource.java:18]: Singleton pattern, the constructor shouldn't be accessible

File: plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        // thread
+        builder.thread(Thread.currentThread());
+        builder.epochMills(SystemClock.now());
+
+        // MDC
+        Map<String, String> contextData = ThreadContext.getImmutableContext();
+        builder.contextData(config.getStringList(LogMapper.MDC_KEYS), contextData);
+
+        // span context
+        builder.spanContext();
```
Comment (zhao-kun) [inline plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java:96]: logInfo().getArgs()[?] could  throw out of index exception, I don't think the code isn't robust, I recommend that: 
```
Args args = logInfo.getArgs()
for (int i = 0 ;i< args.length; ++i) {
   switch i {
     case 0:
       build.set(args[i]....)
       break;
     case 1:  
       build.set(args[i]....)
       break;
     case 2:
       build.set(args[i]....)
       break;
....
   }
return build.build();
}
```

Comment (Oseenix) [reply plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java:96]: The enhanced method always has 5 parameters, it never causes out of index exception.

File: plugin-api/src/main/java/io/opentelemetry/sdk/resources/EaseAgentResource.java (hunk 1 above)
Comment (Oseenix) [reply plugin-api/src/main/java/io/opentelemetry/sdk/resources/EaseAgentResource.java:18]: OK

File: plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java (hunk 2 above)
Comment (zhao-kun) [reply plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java:96]: Even though it always has 5 arguments,  no one can ensure the argument will not be changed in the future. This is is the defensive programming style and it does its best to try to promote your code robustness.

Comment (Oseenix) [reply plugins/log4j2-log-plugin/src/main/java/com/megaease/easeagent/log4j2/log/Log4jLogMapper.java:96]: OK, I will improve it.

**Review by zhao-kun — APPROVED** [review]

**Review by akwei — APPROVED** [review]

**Review by landyking — APPROVED** [review]

**Review by observeralone — APPROVED** [review]

**Pull request merged**